marconi-sidechain/README.md

```diff
@@ -198,147 +249,473 @@ As a consequence, if the last chainpoint of the utxo indexer can, at most,
+```JSON
+{
+  "$schema": "http://json-schema.org/draft-04/schema#",
+  "type": "object",
+  "properties": {
+    "jsonrpc": {
+      "type": "string"
+    },
+    "method": {
+      "type": "string"
+    },
+    "params": {
+      "type": "string"
+    },
+    "id": {
+      "type": "integer"
+    }
+  },
+  "required": [
+    "jsonrpc",
+    "method",
+    "params",
+    "id"
+  ]
+}
+```
 
-**Example**:
+**JSON Schema response**
 
-```sh
-$ curl -d '{"jsonrpc": "2.0" , "method": "getCurrentSyncedPoint" , "params": "", "id": 1}' -H 'Content-Type: application/json' -X POST http://localhost:3000/json-rpc | jq
+```JSON
 {
-  "id": 1,
-  "jsonrpc": "2.0",
-  "result": {
-      "blockHeaderHash": "6161616161616161616161616161616161616161616161616161616161616161",
-      "slotNo": 1,
-      "tag": "ChainPoint"
-  }
+  "$schema": "http://json-schema.org/draft-04/schema#",
+  "type": "object",
+  "properties": {
+    "id": {
+      "type": "integer"
+    },
+    "jsonrpc": {
+      "type": "string"
+    },
+    "result": {
+      "type": "object",
+      "properties": {
+        "blockNo": {
+          "_comment": "Not available yet",
+          "type": "integer"
+        },
+        "blockTimestamp": {
+          "_comment": "Not available yet",
+          "type": "string"
+        },
+        "blockHeaderHash": {
+          "type": "string"
+        },
+        "slotNo": {
+          "type": "integer"
+        },
+        "epochNo": {
+          "_comment": "Not available yet",
+          "type": "integer"
+        }
+      },
+      "required": [
+        "blockNo",
+        "blockTimestamp"
+      ]
+    }
+  },
+  "required": [
+    "id",
+    "jsonrpc",
+    "result"
+  ]
 }
+```
 
-OR
+**Example**:
 
+```sh
+$ curl -d '{"jsonrpc": "2.0" , "method": "getCurrentSyncedBlock" , "params": "", "id": 1}' -H 'Content-Type: application/json' -X POST http://localhost:3000/json-rpc | jq
 {
   "id": 1,
   "jsonrpc": "2.0",
   "result": {
-    "tag": "ChainPointAtGenesis"
+      "blockHeaderHash": "6161616161616161616161616161616161616161616161616161616161616161",
+      "slotNo": 1
   }
 }
 ```
 
-#### getUtxoFromAddress
+#### getUtxosFromAddress (PARTIALLY IMPLEMENTED)
 
 Retrieves UTXOs of a given address until a given point in time (measured in slots).
 
-**Parameters**:
+**JSON Schema request body**:
 
-* `address`: address encoded in the Bech32 format
-* `slotNo`: slot number.
+```JSON
+{
+  "$schema": "http://json-schema.org/draft-04/schema#",
+  "type": "object",
+  "properties": {
+    "jsonrpc": {
+      "type": "string"
+    },
+    "method": {
+      "type": "string"
+    },
+    "params": {
+      "type": "object",
+      "properties": {
+        "address": {
+          "type": "string",
+          "_comment": "Address encoded in the Bech32 format"
+        },
+        "createdAfterSlotNo": {
+          "_comment": "Not available yet",
+          "type": "integer"
+        },
+        "unspentBeforeSlotNo": {
+          "type": "integer"
```

Comment:
Regarding the use of slot number alone instead of the whole chain point, are we sure that we can handle rollback in a safe way in this scenario?
I think it should be ok, but I didn't put many thoughts in it at the moment.
The main issue being that we may send inconsistent results for two different request if a rollback of 1 and a new push were done between two requests on the same slot.

Reply:
In case of a rollback, we can't do anything about that imo. Inconsistent results should be handled by the user itself. How would changing the type of `unspentBeforeSlot` to a ChainPoint help the user?